### ISSUE TITLE
Missing Compatibility With CustomCrafting

### ISSUE THREAD

**Issue by ghost** [Aeltumn/automated_crafting#3]
> These autocrafters are easy and cheap to make and they can craft every single item with every single possible recipe. (including those added by plugins!)

Unfortunately, it doesn't seem to pick up items with recipes registered via the plugin CustomCrafting.

I understand that development work isn't easy, so I'll be patient!

I just hope that maybe compatibility can be added as my server has many, many custom recipes created through that plugin and we would love to be able to autocraft them.

Thank you for your time!

_P.S. Other than that, this plugin is really cool! Nice work!_

**Comment by Aeltumn** [comment]
This seems to be on CustomCrafting's end, bukkit isn't good at recipes so custom crafting chose to implement the recipes by modifying how the crafting table works instead of telling bukkit about new recipes. This makes automated crafting unable to scan the recipes. I'll add compatibility specifically for CustomCrafting then.

Let me know if you encounter any issues with the custom crafting recipes when using autocrafters.

**Comment by Aeltumn** [comment]
I've also clarified the description on spigotmc, it's misleading to say all plugins work. It's just that bukkit is really bad at making recipes customisable so there's no standard way for plugins to communicate when they've got custom recipes making it impossible to create a one size fits all solution for recipe compatibility.

**Comment by ghost** [comment]
Sorry for the late reply, but thank you for getting back to me, and adding support despite the different way CC handles recipes!

You're a nice dev, and that's appreciated. :)

**Comment by ghost** [comment]
Unfortunately, I'm still unable to autocraft many custom recipes from CC.

Here's my use case:
- I have a custom item created with CC called "Compressed Flesh Block" (we'll call it a CF block from now on for simplicity).
- The crafting recipe for a CF block is simply 9 rotten flesh.
- I setup the autocrafter as I am supposed to and place in the block... "Autocrafter can't craft this item".

It does however understand CC recipes using only default items. I REALLY would need it to also support custom items if possible.

I appreciate your patience with this Aeltumn. c:

**Comment by Aeltumn** [comment]
> 
> 
> Unfortunately, I'm still unable to autocraft many custom recipes from CC.
> 
> Here's my use case:
> 
>     * I have a custom item created with CC called "Compressed Flesh Block" (we'll call it a CF block from now on for simplicity).
> 
>     * The crafting recipe for a CF block is simply 9 rotten flesh.
> 
>     * I setup the autocrafter as I am supposed to and place in the block... "Autocrafter can't craft this item".
> 
> 
> It does however understand CC recipes using only default items. I REALLY would need it to also support custom items if possible.
> 
> I appreciate your patience with this Aeltumn. c:

I'll keep at this until I get the compatibility right, so no worries. I'll reopen this and look into how custom items work in a few days.
~~Anyways, if your CF block is made up of 9 rotten flesh, why doesn't that work, isn't rotten flesh a normal item?~~

EDIT: Just realised you mean it doesn't work if the result item is a custom item, I'll take a look at it! I thought you meant ingredients not working as custom items, which is probably also the case.

**Comment by ghost** [comment]
Hi Aeltumn! I understand if you're busy - I am just wondering how things are going!

I appreciate you taking this on in the first place, so I will be patient :)

**Comment by Aeltumn** [comment]
I'm taking a look at this but I don't know how soon I'll be able to add custom item support. Automated Crafting was originally meant to be as purely vanilla as possible but CustomCrafting's custom items are fundamentally not vanilla. There's no easy clean solution to add CustomCrafting support without some major changes.

**Comment by ghost** [comment]
Totally understood! Whilst I've never worked with development of minecraft plugins, etc. I have had programming experience and understand how your implementation could totally not line up with the way CC implements recipes and custom items, and how that could take a lot of work and time.

TL;DR, I'll look forward to it, but won't expect you to do it instantly :P 

Thanks for getting back to me!

**Comment by Aeltumn** [comment]
I've finally gotten around to working on Automated Crafting and I've rewritten a bit of the internals of the plugin to not parse items to json. (which caused problems with customcrafting as the custom-ness disappears in the json format) I'm not certain if this allows you to use custom items but I will test it before releasing the next version and update this comment if it is indeed fixed.

UPDATE: I can confirm crafting recipes with custom items as result work in the newest release. I'm sorry it took a while to get this fixed but I hope you still have use for it! This feature will be in the next update.

**Comment by ghost** [comment]
Awesome! Thank you so much Aeltumn. 😄
